forge-ai/src/main/java/forge/ai/ComputerUtilCard.java

```diff
@@ -875,12 +875,29 @@ public static String getMostProminentType(final CardCollectionView list, final C
         int max = 0;
         String maxType = "";
 
-        for (final Entry<String, Integer> entry : typesInDeck.entrySet()) {
-            final String type = entry.getKey();
+        // Check if the valid list is not empty
+        if (!valid.isEmpty()) {
+            // Iterate through typesInDeck and consider only valid types
+            for (final Entry<String, Integer> entry : typesInDeck.entrySet()) {
+                final String type = entry.getKey();
+
+                // Only consider the types that are in the valid list
+                if (valid.contains(type)) {
```

Comment:
you can save a lot of lines if you just check `valid.isEmpty() ||` here first instead

Reply:
I'll work it out 😅